packages/table/src/table.tsx

```diff
@@ -506,6 +529,8 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         this.setState({
             columnWidths: newColumnWidths,
             focusedCell: enableFocus ? newFocusedCellCoordinates : undefined,
+            numFrozenColumnsClamped: this.getNumFrozenColumnsClamped(nextProps),
```

Comment:
I like the idea of a private function too. But really I'd prefer to just get it from `this.props`, hence my approach of throwing an error before. :/